src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,104 @@
 package core.basesyntax;
 
+import static java.lang.Integer.parseInt;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+
 public class WorkWithFile {
+    static final int MAX_LINES = 100;
 
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] rawFields = readToTable(fromFileName);
+        String[] sumFields = sumTogether(rawFields);
+        saveToFile(sumFields, toFileName);
+    }
+
+    public String [] readToTable(String fileName) {
+        String [] temporaryFields = new String[MAX_LINES];
+        int count = 0;
+        File file = new File(fileName);
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(file));
+            String value = reader.readLine();
+
+            while (value != null) {
+                temporaryFields[count] = value;
+                count++;
+                value = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot read", e);
+        }
+        //CLEAN FROM NULLS
+        String [] resultingFields = new String [count];
+        for (int i = 0; i < count; i++) {
+            resultingFields [i] = temporaryFields[i];
+        }
+        Arrays.sort(resultingFields, Collections.reverseOrder());
+        return resultingFields;
+    }
+
+    public String [] sumTogether(String [] dataFields) {
+        int currentId = 0;
+        String [] temporaryFields = new String[dataFields.length];
+        for (int i = 0; i < dataFields.length; i++) {
+            int isIt = getExistingIndex(temporaryFields, dataFields[i].split(",")[0]);
+            if (isIt >= 0) {
+                temporaryFields[isIt] = dataFields[i].split(",")[0] + ","
+                        + (parseInt(temporaryFields[isIt].split(",")[1])
+                        + parseInt(dataFields[i].split(",")[1]));
+            } else {
+                temporaryFields[currentId] = dataFields[i];
+                currentId++;
+            }
+        }
```

Comment:
This logic looks very complicated, let's create 2 numbers buy and supply, go through all the records and see if there is buy or supply string inside and add values to the appropriate variables